mne/io/meas_info.py

```diff
@@ -186,6 +224,68 @@ def set_montage(self, montage, match_case=True, match_alias=False,
         return self
 
 
+class ContainsMixin(object):
+    """Mixin class for Raw, Evoked, Epochs and Info."""
+
+    def __contains__(self, ch_type):
+        """Check channel type membership.
+
+        Parameters
+        ----------
+        ch_type : str
+            Channel type to check for. Can be e.g. 'meg', 'eeg', 'stim', etc.
+
+        Returns
+        -------
+        in : bool
+            Whether or not the instance contains the given channel type.
+
+        Examples
+        --------
+        Channel type membership can be tested as::
+
+            >>> 'meg' in inst  # doctest: +SKIP
+            True
+            >>> 'seeg' in inst  # doctest: +SKIP
+            False
+
+        """
+        info = self if isinstance(self, Info) else self.info
+        if ch_type == 'meg':
+            has_ch_type = (_contains_ch_type(info, 'mag') or
+                           _contains_ch_type(info, 'grad'))
+        else:
+            has_ch_type = _contains_ch_type(info, ch_type)
+        return has_ch_type
+
+    @property
+    def compensation_grade(self):
+        """The current gradient compensation grade."""
+        info = self if isinstance(self, Info) else self.info
+        return get_current_comp(info)
+
+    @fill_doc
+    def get_channel_types(self, picks=None, unique=False, only_data_chs=False):
+        """Get a list of channel type for each channel.
+
+        Parameters
+        ----------
+        %(picks_all)s
+        unique : bool
+            Whether to return only unique channel types. Default is ``False``.
+        only_data_chs : bool
+            Whether to ignore non-data channels. Default is ``False``.
+
+        Returns
+        -------
+        channel_types : list
+            The channel types.
+        """
+        info = self if isinstance(self, Info) else self.info
+        return _get_channel_types(info, picks=picks, unique=unique,
+                                  only_data_chs=only_data_chs)
```

Comment:
I would rather not copy-paste this from `channels/channels.py` to this one. It makes `git blame` harder with limited maintainability benefit (which I'm not totally convinced about here anyway)

Reply:
It's to avoid a circular import which since it's a class, you have to move it. You don't think it would be more helpful to have the `ContainsMixin` behavior for `Info`? I'm just thinking ahead because I needed the `get_montage`. It seems like operations on just the `info` should be accessible via the info (i.e. if you don't need access to things in the `raw`/`epochs`/`evoked`)

Reply:
Ahh okay if it solves a circular import issue that's fine.

> You don't think it would be more helpful to have the ContainsMixin behavior for Info?

Sure, it's fine to have this (outside of changing the existing behavior of `Info.__contains__`, which fortunately this PR doesn't actually do)